### ISSUE TITLE
Workspace - Settings -  Can't select currency item text

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#29685]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.3.84.3
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @Krishna2323 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1697421082877669

## Action Performed:
1. Open settings > workspaces > select workspace > settings > default currency
2. Try to select text of any currency item

## Expected Result:
ext should be selectable.

## Actual Result:
Text is not selectable.

## Workaround:
Unknown 

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos
<details>
<summary>Android: Native</summary>

 

https://github.com/Expensify/App/assets/93399543/bd879051-4ee4-44cf-92c4-9b819440d8bc

</details>

<details>
<summary>Android: mWeb Chrome</summary>

</details>

<details>
<summary>iOS: Native</summary>

</details>

<details>
<summary>iOS: mWeb Safari</summary>

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

 

https://github.com/Expensify/App/assets/93399543/8d224bf6-4f30-4345-b58b-70a417346db6

https://github.com/Expensify/App/assets/93399543/08115047-3792-401a-a86c-6a65bbe70ac6

</details>

<details>
<summary>MacOS: Desktop</summary>

 

https://github.com/Expensify/App/assets/93399543/26dcaeb7-a7d1-411c-ac56-523db4346a05

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by eh2077** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

Can't select currency item text

### What is the root cause of that problem?

The root cause of this issue is that the workspace currency setting select page use `SelectionList`

https://github.com/Expensify/App/blob/ebf8306fe2829f6292bf5245f96829b1e31483a5/src/pages/workspace/WorkspaceSettingsCurrencyPage.js#L90-L99

and the `BaseListItem` has `select-none` style

https://github.com/Expensify/App/blob/ebf8306fe2829f6292bf5245f96829b1e31483a5/src/components/SelectionList/BaseListItem.js#L51

So, we can select currency item text from workspace setting currency page.

But from the `Select a currency` page of request money flow, we're using `OptionsSelector` to display the currency list

https://github.com/Expensify/App/blob/ebf8306fe2829f6292bf5245f96829b1e31483a5/src/pages/iou/IOUCurrencySelection.js#L142-L154

which is allowed to select text.

### What changes do you think we should make in order to solve the problem?

As the `SelectionList` is newer, I think we can let `Select a currency` page use `SelectionList` as well. Then we can have consistent behaviour. 

If we want the use to select text, then we can consider removing `select-none` style from `BaseListItem`. 

### What alternative solutions did you explore? (Optional)

N/A

**Comment by eh2077** [comment]
@Christinadobrzyn I found that selection lists use the new implementation of `SelectionList`, like Workspace currency setting, Pronouns, Timezone, Year and Country, are all not allowed to select text of items. So the IOU currency selection list can also be same with them right? 

cc @pecanoro @narefyev91

**Comment by Christinadobrzyn** [comment]
Hi @eh2077 I'm not quite sure the answer to that. I think that is true but @pecanoro and @narefyev91 would probably know more about the coding.

**Comment by Christinadobrzyn** [comment]
I can reproduce this but I don't know why we would want to be able to highlight a currency (ex. drag the cursor over a currency to highlight it). The currency is designed to be clicked/tapped - not highlighted. 

I don't think this is a bug or something to fix. 

@narefyev91 and @pecanoro what are your thoughts?

**Comment by eh2077** [comment]
> * Although with manual requests, you can highlight the currency so this is a little inconsistent between the two methods to Request money (as mentioned in the chat with @Krishna2323  https://expensify.slack.com/archives/C049HHMV9SM/p1697567221998929?thread_ts=1697421082.877669&cid=C049HHMV9SM)
> 

@Christinadobrzyn Thanks for your input! 

Yes, It's clear we only need to fix the inconsistency of manual requests. C+ team @narefyev91 suggested same in their comment https://github.com/Expensify/App/issues/29685#issuecomment-1765801466

**Comment by narefyev91** [comment]
@Christinadobrzyn yup the main issue here that we have inconsistence between currency selection in workspace setting and request money flows. We need to following the same way for both. In case that new implementation used SelectionList - my suggestion only to update it for currency selection, and do not touch any other places

**Comment by pecanoro** [comment]
@narefyev91 I agree with you! Assigning @eh2077 so we can keep both consistent

**Comment by Christinadobrzyn** [comment]
Payouts due:

Issue Reporter: $50 @Krishna2323 (in Upwork)
Contributor: $500 + $250 @eh2077 
Contributor+: $500 + $250 @narefyev91 

Eligible for 50% #urgency bonus? Y - looks like the PR was created on Oct 18th and merged on Oct 19

Upwork job is [here](https://www.upwork.com/jobs/~013277c25acbd7d5f2).

@eh2077 can you create a regression test?

**Comment by Christinadobrzyn** [comment]
nudge @eh2077 for a regression test when possible! Thanks!

**Comment by eh2077** [comment]
Regression test
1. Click FAB and choose Request money
2. Select Manual tab
3. Click to change the default currency
4. Verify that `Select a currency` page is open and the default currency is highlighted and has a green check mark on the right side
5. On web or desktop, try to drag and select currency item text
6. Verify that the text is not selectable
7. Search a currency `USD` and select it
8. Verify the page is redirected to the `Request money` page with currency changed to `USD`
9. Enter amount and split money with someone
10. Verify that it's able to split the amount money in selected currency with someone

**Comment by eh2077** [comment]
@Christinadobrzyn Sorry for late. Added regression test 👆

**Comment by Christinadobrzyn** [comment]
Awesome! Thanks @eh2077! Created the regression test and paid out based on [this](https://github.com/Expensify/App/issues/29685#issuecomment-1785885354) payment structure. Closing!
